src/fetchers.ts

```diff
@@ -337,21 +360,8 @@ export async function fetchVersion(
     throw new Error(`No versions found`);
   }
 
-  const validVersions = versions.flatMap(({ main, app }) => {
-    const parsedVersion = semver.coerce(main);
-    if (parsedVersion === null) {
-      return [];
-    }
-    if (parsedVersion.prerelease.length > 0 && !spec.prerelease) {
-      return [];
-    }
-
-    const version: Version = {
-      main: parsedVersion.toString(),
-      app: semver.valid(app ?? null) ?? undefined,
-    };
-
-    return version;
+  const validVersions = versions.filter(({ prerelease }) => {
+    return spec.prerelease || !prerelease;
```

Comment:
This is pretty much the whole point of the `flatMap`, filtering out non-semver was somewhat unintentional.

Reply:
Would this be clearer as
```typescript
const validVersions = spec.prerelease
 ? versions 
 : versions.filter(({ prerelease }) => !prerelease);
```

If `spec.prerelease` means "allow prereleases".